src/ShareRoutes.jsx

```diff
@@ -94,3 +73,95 @@ export default function ShareRoutes({installPrefix, appPrefix}) {
     </Routes>
   )
 }
+
+
+/**
+ * Forward page from /share to /share/v/p per spect at:
+ *   https://github.com/bldrs-ai/Share/wiki/URL-Structure
+ * @param {string} appPrefix The install prefix, e.g. /share.
+ * @return {Object}
+ */
+function Forward({appPrefix}) {
+  const location = useLocation()
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    if (location.pathname == appPrefix) {
+      const dest = appPrefix + '/v/p'
+      debug().log('ShareRoutes#useEffect[location]: forwarding to: ', dest)
+      navigate(dest)
+    }
+  }, [location, appPrefix, navigate])
+
+  return <Outlet />
+}
+
+
+/**
+ * Check if input is a url
+ * @param {input} input
+ * @return {boolean} return true if url is found
+ */
+export function looksLikeLink(input) {
+  assertDefined(input)
+  return input.includes('/') || input.endsWith('.ifc')
+}
+
+
+/**
+ * Look for any obvious problems with the given url.
+ * @param {Object} urlStr
+ * @return {boolean} return true if url is found
+ */
+export function trimToPath(urlStr) {
+  assertDefined(urlStr)
+  urlStr = urlStr.trim()
+  if (urlStr.startsWith('http://')) {
+    urlStr = urlStr.substring('http://'.length)
+  } else if (urlStr.startsWith('https://')) {
+    urlStr = urlStr.substring('https://'.length)
+  }
+  const firstSlashNdx = urlStr.indexOf('/')
+  if (firstSlashNdx == -1) {
+    return undefined
+  }
+  return urlStr.substring(firstSlashNdx)
+}
+
+
+/** Named capture groups for a GitHub URL's path parts. */
+const pathParts = [
+  '(?<org>[^/]+)',
+  '(?<repo>[^/]+)',
+  '(?:(?<isBlob>blob)/)?(?<branch>[^/]+)',
+  '(?<file>.+)',
+]
+
+
+/** Matches strings like '/org/repo/branch/dir1/dir2/file.ifc' */
+const re = new RegExp(`^(?:.*com)?/${pathParts.join('/')}$`)
+
```

Comment:
this is cool!

Reply:
I even went simpler!  Now the caller needs to trim the string to just the path first, and then call the regex.  So in the end it was possible to split the work into a few methods.